### PR TITLE
convert : fix set_vocab_sentencepiece

### PR DESCRIPTION
**Pull request by ggerganov**
Using the same logic as in https://github.com/ggerganov/llama.cpp/pull/6852 for `_set_vocab_sentencepiece()`

### CODE REVIEW

**Comment by ggerganov** [comment]
@JustinLin610 Can you five this PR a try using codeqwen and confirm that it continues to convert correctly?

**Comment by ggerganov** [comment]
This change should be good, right?

File: convert-hf-to-gguf.py [hunk 1]
```diff
@@ -523,21 +527,23 @@ def _set_vocab_sentencepiece(self):
...
-
                 for key in added_tokens_json:
-                    key = key.encode("utf-8")
-                    if key not in tokens:
-                        tokens.append(key)
-                        scores.append(-1000.0)
-                        toktypes.append(SentencePieceTokenTypes.USER_DEFINED)
+                    token_id = added_tokens_json[key]
+                    if (token_id >= vocab_size):
+                        print(f'ignore token {token_id}: id is out of range, max={vocab_size - 1}')
```
Comment (ggerganov) [inline convert-hf-to-gguf.py:541]:
```suggestion
                        logger.warning(f'ignore token {token_id}: id is out of range, max={vocab_size - 1}')
```

**Review by mofosyne — APPROVED** [review]
checked that the logic checks out.

Looks like a refactoring kind of bugfix? If so then let's merge it in.

**Pull request merged**